lib/model.js

```diff
@@ -1412,6 +1412,13 @@ module.exports = (function() {
    * @return {Promise<undefined>}
    */
   Model.prototype.destroy = function(options) {
+    var self = this
+      , instances;
+
+    if (!options || !(options.where || options.truncate)) {
```

Comment:
Understandable @bulkan.
Actually one could argue that this if should only be in effect if `options` is defined.

If `destroy` is called with no options at all there's no chance that people are inadvertently using the old where syntax, it's pretty clear they want a full destroy.